cdc/sink/dispatcher/topic/expression_test.go

```diff
@@ -0,0 +1,234 @@
+			table:      "",
+			wantErr:    "invalid topic expression",
+			expected:   "",
+		},
+		{
+			name:       "invalid expression not containing '{schema}', with no prefix and suffix",
+			expression: "{xxx}",
+			schema:     "abc",
+			table:      "",
+			wantErr:    "invalid topic expression",
+			expected:   "",
+		},
+		{
+			name:       "invalid expressions not containing '{schema}', with prefix",
+			expression: "abc{sch}",
+			schema:     "def",
+			table:      "",
+			wantErr:    "invalid topic expression",
+			expected:   "",
+		},
+		{
+			name:       "valid expression containing '{schema}_{table}'",
+			expression: "{schema}_{table}",
+			schema:     "hello",
+			table:      "world",
+			expected:   "hello_world",
+		},
+		{
+			name:       "valid expression containing '{schema}_{table}', schema/table are converted to lower case letters",
+			expression: "{schema}_{table}",
+			schema:     "HELLO",
+			table:      "WORLD",
+			expected:   "hello_world",
+		},
+		{
+			name:       "valid expression containing '{schema}_{table}', the kafka disallowed characters in table are replaced",
+			expression: "{schema}_{table}",
+			schema:     "hello",
+			table:      "!@#$%^&*",
+			expected:   "hello_________",
+		},
+		{
+			name:       "valid expression containing '{schema}_{table}', the kafka disallowed characters in schema are replaced",
+			expression: "{schema}_{table}",
+			schema:     "()_+.{}",
+			table:      "world",
+			expected:   "____.___world",
+		},
+		{
+			name:       "valid expression containing '{schema}_{table}', the kafka disallowed characters in schema and table are replaced",
+			expression: "{schema}_{table}",
+			schema:     "你好",
+			table:      "世界",
+			expected:   "_____",
+		},
+		{
+			name:       "invalid expression not containing '{schema}_{table}'",
+			expression: "{sch}_{tab}",
+			schema:     "hello",
+			table:      "world",
+			wantErr:    "invalid topic expression",
+			expected:   "",
+		},
+		{
+			name:       "invalid topic name '.'",
+			expression: "{schema}",
+			schema:     ".",
+			table:      "",
+			expected:   "_",
+		},
+		{
+			name:       "invalid topic name '..'",
+			expression: "{schema}",
+			schema:     "..",
+			table:      "",
+			expected:   "__",
+		},
+	}
+
+	for _, tc := range cases {
+		topicExpr := Expression(tc.expression)
+		err := topicExpr.Validate()
+		if tc.wantErr != "" {
+			require.Contains(t, err.Error(), tc.wantErr, fmt.Sprintf("case:%s", tc.name))
+		} else {
+			require.Nil(t, err, fmt.Sprintf("case:%s", tc.name))
+			topicName := topicExpr.Substitute(tc.schema, tc.table)
+			require.Equal(t, topicName, tc.expected, fmt.Sprintf("case:%s", tc.name))
+		}
+	}
+}
+
+// goos: linux
+// goarch: amd64
+// pkg: github.com/pingcap/tiflow/cdc/sink/dispatcher
+// cpu: Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
+// BenchmarkSubstitute/schema_substitution-40         	  199372	      6477 ns/op
+// BenchmarkSubstitute/schema_table_substitution-40   	  110752	     13637 ns/op
```

Comment:
It is better to add the execution command to the comment.

Reply:
> It is better to add the execution command to the comment.

OK, I will add the command.